### DISCUSSION TITLE
How to add a tab button to the right hand side (accordion-aside)?

### DISCUSSION THREAD

**Discussion by samuel25gh** [anapnoe/stable-diffusion-webui-ux#215] [Q&A] [answered]
Hi. I would like to make a tab button of my own. As I attempt to do so I come across some problems:

1. I've managed to add a new tab button on the left by adding a new interface in the modules/ui.py . However, I would like to make a new tab button on the right, but I can't seem to find any way to do so.

2. I would also like to change the position of some installed extension tab buttons from be on the left to the right but I just don't know where to modify the code. I've tried setting the ui_tab_order but it has no effect at all. It would be wonderful if I can get to know where in the project code controls the sidebar on the right. 

![image](https://github.com/anapnoe/stable-diffusion-webui-ux/assets/90382562/2568d1a1-5a00-4cf0-b0cf-eaf8a55fbfe6)

**Comment by anapnoe** [comment, marked as answer]
Hi you would be able to do that with workspaces and have as many layout variations as you like
the modules are loaded dynamically and IN and MY are not part of the UI/UX layout
If you want to do the mod for your specific use case the function is named `createButtons4Extensions()` located in `anapnoe_sd_uiux_core.js ` file. I am guessing that you want to move the contents of the module also to the aside area thus you will need to look into the
layout file `template-aside-extra-networks.html` 
you can study the `anapnoe-sd-theme-editor` extension to see how it is hooked to the aside view
the downside of this is every time you update you need to redo the changes so keep a backup of your mod files

**Reply by samuel25gh** [reply]
Thank you for answering! 

I added code in the template-aside-extra-networks.html and was able to make a button being shown on the right. 

But this also brings me more questions as I tried to do the same on template-app-root.html to add a tab button at the bottom-left (where Sponsors & Supporters, Contributors tabs are located). Even I added code on the template-app-root.html the changes didn't show like it did for template-aside-extra-networks.html. Could you please explain to me why that is and how to add a tab button to the bottom-left? Thank you.

![image](https://github.com/anapnoe/stable-diffusion-webui-ux/assets/90382562/f3c2889f-55ab-4656-b2c6-7704a6488414)
